src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -0,0 +1,914 @@
+        /// or the condition being false would cause the method to return false</param>
+        /// <param name="remainingStatements">Potential remaining statements of the branch that does not return false</param>
+        /// <returns>whether the pattern was matched (one of the branches must have a simple "return false")</returns>
+        private static bool TryGetSuccessCondition(
+            IOperation whenTrue,
+            IOperation? whenFalse,
+            ImmutableArray<IOperation> otherOps,
+            out bool successRequirement,
+            out ImmutableArray<IOperation> remainingStatements)
+        {
+            // this will be changed if we successfully match the pattern
+            successRequirement = default;
+            // this could be empty even if we match, if there is no else block
+            remainingStatements = default;
+
+            // all the operations that would happen after the condition is true or false
+            // branches can either be block bodies or single statements
+            // each branch is followed by statements outside the branch either way
+            var trueOps = ((whenTrue as IBlockOperation)?.Operations ?? ImmutableArray.Create(whenTrue))
+                .Concat(otherOps);
+            var falseOps = ((whenFalse as IBlockOperation)?.Operations ??
+                (whenFalse != null
+                    ? ImmutableArray.Create(whenFalse)
+                    : ImmutableArray.Create<IOperation>()))
+                .Concat(otherOps);
+
+            // We expect one of the true or false branch to have exactly one statement: return false.
+            // If we don't find that, it indicates complex behavior such as
+            // extra statments, backup equality if one condition fails, or something else.
+            // We don't necessarily expect a return true because we could see
+            // a final return statement that checks a last set of conditions such as:
+            // if (other is C otherC)
+            // {
+            //     return otherC.A == A;
+            // }
+            // return false;
+            // We should never have a case where both branches could potentially return true;
+            // at least one branch must simply return false.
+            if (ReturnsFalseImmediately(trueOps) == ReturnsFalseImmediately(falseOps))
+                // both or neither fit the return false pattern, this if statement either doesn't do
+                // anything or does something too complex for us, so we assume it's not a default.
+                return false;
+
+            // when condition succeeds (evaluates to true), we return false
+            // so for equality the condition should not succeed
+            successRequirement = !ReturnsFalseImmediately(trueOps);
+            remainingStatements = successRequirement ? trueOps : falseOps;
+            remainingStatements.Concat(otherOps);
```

Comment:
This is fishy: remainingStatements is immutable, so this isn't doing anything.

Reply:
Yes, fortunately we made that concat call to both `trueOps` and `falseOps` before they were assigned to `remainingStatements`, so `otherOps` still got added to the end, even though this line does nothing.